selfdrive/ui/qt/offroad/wifiManager.cc

```diff
@@ -392,6 +381,37 @@ void WifiManager::disconnect() {
   }
 }
 
+QString WifiManager::ssid_from_path(const QDBusObjectPath &path) {
+  QDBusInterface nm2(nm_service, path.path(), nm_settings_conn_iface, bus);
+  nm2.setTimeout(dbus_timeout);
+
+  QDBusMessage response = nm2.call("GetSettings");
+  const QDBusArgument &dbusArg = response.arguments().at(0).value<QDBusArgument>();
+
+  QString ssid;
+  QMap<QString, QMap<QString,QVariant>> map;
+  dbusArg >> map;
+  for (auto &inner : map) {
+    for (auto &val : inner) {
+      QString key = inner.key(val);
+      if (key == "ssid") {
+        return val.toString();
+      }
+    }
+  }
+  return "";
+}
+
+QDBusObjectPath WifiManager::path_from_ssid(const QString &ssid) {  // ensure ssid is a known connection before calling this function
+  for (const QDBusObjectPath path : list_connections()) {
+    if (ssid_from_path(path) == ssid) {
+      return path;
+    }
+  }
+  QDBusObjectPath path;
+  return path;
```

Comment:
Not sure what to do here, you should only call this function if it's a known connection, and every referencing location does check that, so is returning an empty QDBusObjectPath variable okay, or something like `NULL`? Not sure what best practice is here.

Reply:
let's just check if the path is empty